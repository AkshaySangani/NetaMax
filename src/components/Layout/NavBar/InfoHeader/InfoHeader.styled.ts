import styled from "styled-components";
import { responsive } from "styled/responsive";

import { Box, Flex } from "@chakra-ui/react";

export const MainInfoHeader = styled(Box)`
  padding: 5px 0px;

  ${responsive.smPlus} {
    padding: 0px;
    width: 100%;
    align-self: center;
  }
`;

export const InfoHeaderContainer = styled(Flex)`
  margin-top: 20px;
  height: 65px;
  border-radius: 0px;
  justify-content: right;
  padding-right: 20px;
  background: rgba(123, 1, 0, 0.42);
  width: 100vw;
  position: relative;
  left: calc(-50vw + 50%);
`;

export const MinifiedInfoHeaderContainer = styled(Flex)`
  justify-content: center;
  background: rgba(123, 1, 0, 0.42);
  height: 30px;
  width: 100vw;
  position: relative;
  left: calc(-50vw + 50%);
`;
