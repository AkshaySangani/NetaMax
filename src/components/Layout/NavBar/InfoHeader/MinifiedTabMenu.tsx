import React, { ReactElement } from "react";

import { MI_CUENTA_ROUTE } from "constants/checkoutConstants";
import router from "next/router";
import BagIcon from "styled/icons/BagIcon";
import IconCoin from "styled/icons/Coin";
import StoreIcon from "styled/icons/StoreIcon";

import { Box } from "@chakra-ui/react";

import { CustomTab, MinifiedTabMenuContainer } from "../TabMenu/TabMenu.styled";

/**
 * The InfoHeader Component
 * @param {IInfoHeaderProps} props the InfoHeader props.
 * @returns {ReactElement} the InfoHeader component.
 */
export const MinifiedTabMenu = (): ReactElement => {
  return (
    <MinifiedTabMenuContainer pt="14px">
      <Box
        background="#FFFFFF"
        borderBottom="none"
        w="100%"
        h="64px"
        display="flex"
        justifyContent="space-between"
        padding="7px"
      >
        <CustomTab
          onClick={() => router.push("/")}
          data-testid="Tiendita tab"
          activeTab={router.pathname === "/"}
          as="button"
        >
          <Box display="flex" flexDirection="column" alignItems="center" justifyContent="center">
            <StoreIcon
              width="22px"
              height="19px"
              fill={router.pathname === "/" ? "#3870FF" : "#5F5F5F"}
            />
            <Box marginTop="5px" marginLeft="4px">
              Tiendita
            </Box>
          </Box>
        </CustomTab>

        <CustomTab
          onClick={() => router.push(MI_CUENTA_ROUTE)}
          activeTab={router.pathname === MI_CUENTA_ROUTE}
          data-testid="Mi cuenta tab"
          as="button"
        >
          <Box display="flex" flexDirection="column" alignItems="center" justifyContent="center">
            <BagIcon
              width="18px"
              height="22px"
              fill={router.pathname === MI_CUENTA_ROUTE ? "#3870FF" : "#5F5F5F"}
            />
            <Box marginTop="5px" marginLeft="4px">
              Mi cuenta
            </Box>
          </Box>
        </CustomTab>

        <CustomTab
          onClick={() => router.push("/gana-dinero")}
          isWinMoneyTab={router.pathname === "/gana-dinero"}
          as="button"
          data-testid="Gana dinero tab"
        >
          <Box display="flex" flexDirection="column" alignItems="center" justifyContent="center">
            <IconCoin width="26px" height="26px" />
            <Box
              marginTop="3px"
              marginLeft="4px"
              color={router.pathname === "/gana-dinero" ? "#ED9F00" : "#5F5F5F"}
            >
              Gana dinero
            </Box>
          </Box>
        </CustomTab>
      </Box>
    </MinifiedTabMenuContainer>
  );
};
