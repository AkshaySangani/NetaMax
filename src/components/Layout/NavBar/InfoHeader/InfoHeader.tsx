import React, { ReactElement } from "react";

import { TabMenu } from "../TabMenu/TabMenu";
import { InfoHeaderContainer, MainInfoHeader } from "./InfoHeader.styled";

/**
 * The InfoHeader Component
 * @param {IInfoHeaderProps} props the InfoHeader props.
 * @returns {ReactElement} the InfoHeader component.
 */
export const InfoHeader = (): ReactElement => {
  return (
    <MainInfoHeader>
      <InfoHeaderContainer>
        <TabMenu />
      </InfoHeaderContainer>
    </MainInfoHeader>
  );
};
