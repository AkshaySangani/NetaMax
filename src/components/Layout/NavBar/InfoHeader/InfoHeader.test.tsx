import { render, cleanup } from "@testing-library/react";
import { InfoHeader } from "./InfoHeader";
import mockRouter from "next-router-mock";

jest.mock("next/router", () => require("next-router-mock"));

describe("Render InfoHeader component", () => {
  beforeEach(() => {
    mockRouter.setCurrentUrl("/");
  });

  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  test("InfoHeader should be render correctly", () => {
    const { container } = render(<InfoHeader />);
    expect(container).toMatchSnapshot();
  });
});
