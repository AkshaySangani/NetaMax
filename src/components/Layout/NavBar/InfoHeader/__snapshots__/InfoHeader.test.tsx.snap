// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render InfoHeader component InfoHeader should be render correctly 1`] = `
<div>
  <div
    class="sc-eCImPb hiwpXB css-0"
  >
    <div
      class="sc-jRQBWg rTWmE css-k008qs"
    >
      <div
        class="sc-bdvvtL khGriJ css-dq90ix"
      >
        <div
          class="tab-container css-75xu7l"
        >
          <button
            class="sc-gsDKAQ cscLow"
            data-testid="Tiendita tab"
          >
            <div
              class="tab-items-container css-u6v8er"
            >
              <svg
                height="19px"
                viewBox="0 0 22 19"
                width="22px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  clip-path="url(#clip0_1845_27211)"
                >
                  <path
                    d="M22 5.93667V6.57417C21.93 6.81323 21.8751 7.05813 21.7847 7.29029C21.3497 8.43831 20.5434 9.20277 19.3145 9.43652C18.0963 9.66708 17.0762 9.25484 16.2758 8.32462C16.1681 8.19978 16.1202 8.22528 16.0282 8.33312C14.7024 9.88436 12.4464 9.88755 11.1297 8.33897C11.0258 8.21678 10.979 8.20297 10.867 8.33472C9.54768 9.88596 7.29008 9.88649 5.97235 8.33472C5.86469 8.20616 5.81463 8.20828 5.70966 8.33472C5.26902 8.88033 4.65274 9.261 3.9629 9.41368C2.53858 9.74571 1.09111 9.05084 0.397254 7.719C0.0861214 7.12082 -0.109815 6.47854 0.0559789 5.80014C0.439781 4.2319 0.846192 2.66898 1.24399 1.10393C1.42324 0.400559 1.9357 0 2.67046 0C8.22492 0 13.7797 0 19.3349 0C20.0568 0 20.573 0.402154 20.7479 1.0933C21.023 2.1728 21.2894 3.25441 21.5699 4.33391C21.7131 4.86887 21.8175 5.41286 22 5.93667Z"
                  />
                  <path
                    d="M1.37744 10.4347C1.37744 12.8254 1.37744 15.1772 1.37744 17.5301C1.37744 18.4284 1.95557 18.9952 2.86851 18.9958C4.59428 18.9958 6.3195 18.9958 8.04526 19C8.20675 19 8.25628 18.965 8.25574 18.796C8.2482 17.1215 8.25574 15.4492 8.24712 13.772C8.24712 13.61 8.28319 13.5638 8.45383 13.5643C10.1509 13.5718 11.8478 13.5718 13.5445 13.5643C13.7092 13.5643 13.7555 13.5999 13.7549 13.7683C13.7479 15.4433 13.7549 17.1178 13.7469 18.7923C13.7469 18.9549 13.7829 19.0006 13.9536 19C15.6648 18.9931 17.376 18.9958 19.0872 18.9958C20.0702 18.9958 20.6246 18.4449 20.6246 17.469V10.4395C20.5476 10.4677 20.4884 10.4884 20.4298 10.5112C18.9943 11.0676 17.62 10.9455 16.307 10.1447C16.1772 10.065 16.1003 10.0948 15.984 10.1596C14.3846 11.0623 12.7831 11.0637 11.1797 10.1638C11.0436 10.0873 10.959 10.0841 10.8212 10.1638C9.22197 11.0669 7.61625 11.0669 6.01806 10.1638C5.87756 10.0841 5.79197 10.0894 5.66009 10.1713C4.82804 10.6972 3.84038 10.9305 2.85721 10.8332C2.34575 10.7782 1.84633 10.6437 1.37744 10.4347Z"
                  />
                  <path
                    d="M1.37746 10.4347C1.84718 10.643 2.34734 10.7766 2.85938 10.8305C3.84309 10.9282 4.83138 10.6949 5.66388 10.1686C5.79576 10.0868 5.8792 10.0814 6.02185 10.1611C7.62004 11.0642 9.22576 11.0642 10.825 10.1611C10.9628 10.0836 11.0473 10.0868 11.1835 10.1611C12.7833 11.0614 14.3848 11.06 15.9878 10.1569C16.1024 10.0921 16.1794 10.0623 16.3108 10.142C17.621 10.9452 18.9952 11.0674 20.4335 10.5086C20.4922 10.4857 20.5514 10.465 20.6284 10.4368V17.4652C20.6284 18.4411 20.074 18.9915 19.091 18.992C17.3798 18.992 15.6686 18.9894 13.9573 18.9963C13.7867 18.9963 13.7501 18.9511 13.7506 18.7886C13.7566 17.1141 13.7506 15.4417 13.7587 13.7646C13.7587 13.5962 13.713 13.56 13.5482 13.5606C11.8515 13.5669 10.1547 13.5669 8.45762 13.5606C8.28698 13.5606 8.25038 13.6063 8.25091 13.7683C8.25737 15.4428 8.25091 17.1151 8.25953 18.7923C8.25953 18.9612 8.2127 18.9968 8.04905 18.9963C6.32329 18.9899 4.59807 18.9931 2.8723 18.992C1.9572 18.992 1.38177 18.4247 1.38123 17.5263C1.37692 15.1771 1.37746 12.8243 1.37746 10.4347Z"
                  />
                </g>
                <defs>
                  <clippath
                    id="clip0_1845_27211"
                  >
                    <rect
                      fill="white"
                      height="19"
                      width="22"
                    />
                  </clippath>
                </defs>
              </svg>
              <div
                class="css-15fzge"
              >
                Tiendita
              </div>
            </div>
          </button>
          <button
            class="sc-gsDKAQ iuPBHE"
            data-testid="En grupo tab"
          >
            <div
              class="tab-items-container css-u6v8er"
            >
              <svg
                fill="none"
                height="22px"
                viewBox="0 0 22 18"
                width="18px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16 17H21V15C21 13.3431 19.6569 12 18 12C17.0444 12 16.1931 12.4468 15.6438 13.1429M16 17H6M16 17V15C16 14.3438 15.8736 13.717 15.6438 13.1429M6 17H1V15C1 13.3431 2.34315 12 4 12C4.95561 12 5.80686 12.4468 6.35625 13.1429M6 17V15C6 14.3438 6.12642 13.717 6.35625 13.1429M6.35625 13.1429C7.0935 11.301 8.89482 10 11 10C13.1052 10 14.9065 11.301 15.6438 13.1429M14 4C14 5.65685 12.6569 7 11 7C9.34315 7 8 5.65685 8 4C8 2.34315 9.34315 1 11 1C12.6569 1 14 2.34315 14 4ZM20 7C20 8.10457 19.1046 9 18 9C16.8954 9 16 8.10457 16 7C16 5.89543 16.8954 5 18 5C19.1046 5 20 5.89543 20 7ZM6 7C6 8.10457 5.10457 9 4 9C2.89543 9 2 8.10457 2 7C2 5.89543 2.89543 5 4 5C5.10457 5 6 5.89543 6 7Z"
                  stroke="#533CDE"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.6"
                />
              </svg>
              <div
                class="css-bbu3jp"
              >
                En Grupo
              </div>
            </div>
          </button>
          <button
            class="sc-gsDKAQ iuPBHE"
            data-testid="Mi cuenta tab"
          >
            <div
              class="tab-items-container css-u6v8er"
            >
              <svg
                height="22px"
                viewBox="0 0 24 24"
                width="22px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M23.1998 12C23.1998 18.1856 18.1854 23.2 11.9998 23.2C5.8142 23.2 0.799805 18.1856 0.799805 12C0.799805 5.8144 5.8142 0.799988 11.9998 0.799988C18.1854 0.799988 23.1998 5.8144 23.1998 12ZM14.7998 7.79999C14.7998 9.34639 13.5462 10.6 11.9998 10.6C10.4534 10.6 9.19979 9.34639 9.19979 7.79999C9.19979 6.25359 10.4534 4.99999 11.9998 4.99999C13.5462 4.99999 14.7998 6.25359 14.7998 7.79999ZM11.9997 13.4C9.17513 13.4 6.74131 15.0729 5.63502 17.482C7.17543 19.2688 9.4555 20.4 11.9998 20.4C14.544 20.4 16.824 19.2689 18.3644 17.4821C17.2581 15.073 14.8243 13.4 11.9997 13.4Z"
                  fill-rule="evenodd"
                />
              </svg>
              <div
                class="css-15fzge"
              >
                Mi cuenta
              </div>
            </div>
          </button>
          <button
            class="sc-gsDKAQ kTbIqS"
            data-testid="Gana dinero tab"
          >
            <div
              class="tab-items-container css-u6v8er"
            >
              <svg
                fill="none"
                height="26px"
                viewBox="0 0 18 18"
                width="26px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  clip-path="url(#clip0_4082_16816)"
                >
                  <path
                    d="M16.0139 8.27946C16.0881 9.61603 15.7642 10.9446 15.0833 12.0971C14.4024 13.2496 13.395 14.1743 12.1885 14.7542C10.982 15.3341 9.63064 15.5433 8.3053 15.3552C6.97996 15.1671 5.74016 14.5902 4.74269 13.6975C3.74522 12.8047 3.03489 11.6363 2.70153 10.3398C2.36816 9.04335 2.42674 7.67716 2.86985 6.414C3.31295 5.15084 4.12069 4.04745 5.19091 3.24336C6.26113 2.43928 7.54576 1.97062 8.88234 1.89665C10.6743 1.79787 12.4322 2.41474 13.7695 3.61165C15.1068 4.80855 15.9141 6.48752 16.0139 8.27946Z"
                    fill="#E8AF2E"
                  />
                  <path
                    d="M14.7828 8.34982C14.8433 9.44468 14.5779 10.5329 14.0199 11.4769C13.462 12.4209 12.6367 13.1783 11.6484 13.6532C10.6601 14.1282 9.55308 14.2994 8.46744 14.1453C7.38179 13.9911 6.36624 13.5185 5.54919 12.7872C4.73214 12.0558 4.1503 11.0987 3.87723 10.0367C3.60416 8.97469 3.65213 7.85558 4.01508 6.82085C4.37802 5.78612 5.03965 4.88226 5.91628 4.22354C6.79291 3.56482 7.84518 3.18084 8.94004 3.12016C9.66712 3.07966 10.395 3.18283 11.0822 3.42377C11.7694 3.6647 12.4024 4.03869 12.945 4.52436C13.4876 5.01002 13.9292 5.59784 14.2445 6.25423C14.5598 6.91062 14.7427 7.62272 14.7828 8.34982Z"
                    fill="#F9CA35"
                  />
                  <path
                    d="M10.3709 7.06817C10.3102 6.6389 9.90068 6.44269 9.13916 6.48385C8.37763 6.525 8.00284 6.74693 8.02421 7.15412C8.02712 7.27195 8.06707 7.38586 8.1384 7.47969C8.20973 7.57351 8.3088 7.64248 8.42155 7.6768C8.67269 7.77718 9.17566 7.8787 9.92626 7.98287C10.6769 8.08704 11.2384 8.29964 11.6017 8.6129C11.7847 8.77232 11.932 8.96856 12.034 9.18879C12.136 9.40903 12.1904 9.64831 12.1936 9.89099C12.2158 10.1858 12.1563 10.481 12.0217 10.7443C11.8871 11.0075 11.6825 11.2285 11.4305 11.383C10.8687 11.7586 10.2131 11.9695 9.53775 11.9919C7.79613 12.081 6.7684 11.4652 6.45455 10.1443L7.89147 9.95293C8.03527 10.6484 8.56982 10.9718 9.49514 10.9229C10.36 10.8778 10.7771 10.5617 10.7464 9.97448C10.7462 9.85789 10.7138 9.74361 10.6528 9.64426C10.5918 9.5449 10.5045 9.46431 10.4006 9.41138C10.1847 9.29371 9.68472 9.18011 8.90077 9.07057C8.11912 8.95865 7.54567 8.75661 7.1846 8.46295C7.01155 8.32741 6.87026 8.15564 6.77065 7.95969C6.67103 7.76374 6.61551 7.54836 6.60799 7.32868C6.59666 7.08024 6.64509 6.83271 6.7492 6.60685C6.85331 6.38099 7.01007 6.1834 7.20633 6.03064C7.63087 5.66209 8.26592 5.4561 9.11148 5.41268C10.7832 5.32437 11.6879 5.83178 11.8255 6.93492L10.3709 7.06817Z"
                    fill="#C38728"
                  />
                  <path
                    d="M8.42766 4.33385L8.49219 5.58398L9.86645 5.51305L9.80192 4.26291L8.42766 4.33385Z"
                    fill="#C38728"
                  />
                  <path
                    d="M8.81438 11.7831L8.87891 13.0332L10.2532 12.9623L10.1886 11.7121L8.81438 11.7831Z"
                    fill="#C38728"
                  />
                </g>
                <defs>
                  <clippath
                    id="clip0_4082_16816"
                  >
                    <rect
                      fill="white"
                      height="13.539"
                      transform="translate(0 4.77148) rotate(-19.6648)"
                      width="14.1767"
                    />
                  </clippath>
                </defs>
              </svg>
              <div
                class="css-15fzge"
              >
                Gana dinero
                
              </div>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
